alt-tab-macos/ui/Application.swift

```diff
@@ -109,11 +111,12 @@ class Application: NSApplication, NSApplicationDelegate, NSWindowDelegate {
                 return
             }
             selectedOpenWindow = cellWithStep(step)
+            let preferredScreen = Screen.getPreferredScreen() // we want the thumbnail computation and panel display use the same screen (in case mouse movements switching screens etc.)
             var workItem: DispatchWorkItem!
             workItem = DispatchWorkItem {
-                if !workItem.isCancelled { self.thumbnailsPanel!.computeThumbnails() }
+                if !workItem.isCancelled { self.thumbnailsPanel!.computeThumbnails(preferredScreen) }
                 if !workItem.isCancelled { self.thumbnailsPanel!.highlightCellAt(step) }
-                if !workItem.isCancelled { self.showCenteredPanel(self.thumbnailsPanel!) }
+                if !workItem.isCancelled { self.showCenteredFrontPanel(self.thumbnailsPanel!, preferredScreen) }
```

Comment:
It took me a while to understand what was happening here. That's really cool that you covered that use-case!

Reply:
Thanks! 

I found it cause if was monitoring how often getPreferredScreen() was called and during the AltTab trigger it got called twice. There could be a time difference between two calls and it would cause problems if the thumbnails are computed for an huge screen (PC monitor) and the panel is then shown an a MacBook monitor (when the user is using the mouse screen and moved the mouse pointer between both method executions).